src/field255.rs

```diff
@@ -0,0 +1,308 @@
+
+    fn neg(self) -> Field255 {
+        let mut temp = [0; 5];
+        fiat_25519_opp(&mut temp, &self.0);
+        let mut output = [0; 5];
+        fiat_25519_carry(&mut output, &temp);
+        Field255(output)
+    }
+}
+
+impl<'a> TryFrom<&'a [u8]> for Field255 {
+    type Error = FieldError;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, FieldError> {
+        Self::try_from_bytes(bytes, false)
+    }
+}
+
+impl From<Field255> for [u8; Field255::ENCODED_SIZE] {
+    fn from(element: Field255) -> Self {
+        let mut array = [0; 32];
+        fiat_25519_to_bytes(&mut array, &element.0);
+        array.reverse();
+        array
+    }
+}
+
+impl From<Field255> for Vec<u8> {
+    fn from(elem: Field255) -> Vec<u8> {
+        <[u8; Field255::ENCODED_SIZE]>::from(elem).to_vec()
+    }
+}
+
+impl Display for Field255 {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let bytes: [u8; 32] = (*self).into();
+        let bigint = BigInt::from_bytes_be(Sign::Plus, bytes.as_slice());
+        <BigInt as Display>::fmt(&bigint, f)
+    }
+}
+
+impl Debug for Field255 {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let bytes: [u8; 32] = (*self).into();
+        f.debug_tuple("Field255").field(&bytes).finish()
+    }
+}
+
+impl Serialize for Field255 {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let bytes: [u8; Self::ENCODED_SIZE] = (*self).into();
+        serializer.serialize_bytes(&bytes)
+    }
+}
+
+impl<'de> Deserialize<'de> for Field255 {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Field255, D::Error> {
+        deserializer.deserialize_bytes(FieldElementVisitor {
+            phantom: PhantomData,
+        })
+    }
+}
+
+impl Encode for Field255 {
+    fn encode(&self, bytes: &mut Vec<u8>) {
+        // Note that VDAF uses big-endian serialization, while fiat-crypto uses little-endian
+        // serialization.
+        let mut value = [0u8; Self::ENCODED_SIZE];
+        fiat_25519_to_bytes(&mut value, &self.0);
+        value.reverse();
+        bytes.extend_from_slice(&value);
+    }
+}
+
+impl Decode for Field255 {
+    fn decode(bytes: &mut Cursor<&[u8]>) -> Result<Self, CodecError> {
+        let mut value = [0u8; Self::ENCODED_SIZE];
+        bytes.read_exact(&mut value)?;
+        Field255::try_from_bytes(&value, false).map_err(|e| {
+            CodecError::Other(Box::new(e) as Box<dyn std::error::Error + 'static + Send + Sync>)
+        })
+    }
+}
+
+impl FieldElement for Field255 {
+    const ENCODED_SIZE: usize = 32;
+
+    fn pow(&self, _exp: u128) -> Self {
+        unimplemented!()
+    }
+
+    fn inv(&self) -> Self {
+        unimplemented!()
+    }
+
+    fn try_from_random(bytes: &[u8]) -> Result<Self, FieldError> {
+        Field255::try_from_bytes(bytes, true)
+    }
+
+    fn zero() -> Self {
+        Field255([0, 0, 0, 0, 0])
+    }
+
+    fn one() -> Self {
+        Field255([1, 0, 0, 0, 0])
```

Comment:
Let's add a test that asserts this is indeed a valid representation of one. (The byte slice rep should equal `[0, ..., 0, 1]`.) It might also be worth asserting that `Field255::one()` subtracted from itself is `Field255::zero()`.

Similarly for `zero()` above.

Reply:
The subtraction check is covered by `field_element_test_common()`.